specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,205 @@
+    OkResponse & SchemaHeaders & SchemaContent & SchemaContentType
+  >;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/?api-version={apiVersion}
+  @summary("List schema versions.")
+  @doc("Gets the list of all versions of one schema.")
+  op listSchemaVersions is resourceOperations.ResourceList<Version>;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/{schemaVersion}?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ Version resource doesn't allow returning non-version response."
+  @summary("Get specific schema versions.")
+  @doc("Gets one specific version of one schema.")
+  @route("/$schemaGroups/{groupName}/schemas/{name}/versions/{schemaVersion}")
+  @get
+  op getSchemaByVersion is Azure.Core.Foundations.Operation<
+    Azure.Core.Foundations.ItemKeysOf<Version>,
+    OkResponse & SchemaHeaders & SchemaContent & SchemaContentType
+  >;
+
+  //  POST https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}:get-id?api-version={apiVersion}
+  @summary("Get ID for existing schema.")
+  @doc("Gets the ID referencing an existing schema within the specified schema group, as matched by schema content comparison.")
+  @action("get-id")
+  op getSchemaIdByContent is resourceOperations.ResourceAction<
+    Schema,
+    {
+      ...SchemaContentType;
+
+      @doc("String representation (UTF-8) of the registered schema.")
+      @body
+      schemaContent: bytes;
+    },
+    SchemaHeaders
+  >;
+
+  //  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-names" "Register operation does not use create/replace naming"
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard createOrReplace operation does not allow passing in binary body to request."
+  @summary("Register new schema")
+  @doc("Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.")
+  @route("/$schemaGroups/{groupName}/schemas/{name}")
+  @put
+  op registerSchema is Azure.Core.Foundations.Operation<
+    Azure.Core.Foundations.ItemKeysOf<Schema> & SchemaContent & SchemaContentType,
+    NoContentResponse<SchemaHeaders>
+  >;
+}
+
+// Models
+
+alias NoContentResponse<T> = TypeSpec.Http.Response<204> & T;
+
+@doc("The content type for given schema. Each schema type has an associated content-type.")
+model SchemaContentType {
+  @doc("The content type for given schema.")
+  @header("Content-Type")
+  contentType: "application/json; serialization=Avro" | "application/json; serialization=json" | "text/plain; charset=utf-8";
```

Comment:
need to update this to the following for the new api version. How would I do this?
```suggestion
  contentType: "application/json; serialization=Avro" | "application/json; serialization=json" | "text/plain; charset=utf-8" | "text/vnd.ms.protobuf";
```

Reply:
add a `union` so an @added decorator can be added to protobuf type